lib/gitlab_shell.rb

```diff
@@ -56,16 +56,29 @@ def exec
   def parse_cmd
     args = Shellwords.shellwords(@origin_cmd)
     @git_cmd = args.first
+    @git_access = @git_cmd
 
     raise DisallowedCommandError unless GIT_COMMANDS.include?(@git_cmd)
 
-    if @git_cmd == 'git-annex-shell'
+    case @git_cmd
+    when 'git-annex-shell'
       raise DisallowedCommandError unless @config.git_annex_enabled?
 
       @repo_name = escape_path(args[2].sub(/\A\/~\//, ''))
 
       # Make sure repository has git-annex enabled
       init_git_annex(@repo_name)
+    when 'git-lfs-authenticate'
+      raise DisallowedCommandError unless args.count >= 2
+      @repo_name = escape_path(args[1])
+      case args[2]
+      when "download"
+         @git_access = "git-upload-pack"
```

Comment:
Use 2 (not 3) spaces for indentation.<br>Prefer single-quoted strings when you don't need string interpolation or special symbols.